### ISSUE TITLE
successfully executed uge process re-running on resume

### ISSUE THREAD

**Issue by stoneg64** [nextflow-io/nextflow#1022]
### Expected behavior and actual behavior

I have a four process workflow.  I am testing how well the Nextflow can resume previously halted workflows.  My workflow successfully completes three of four tasks.  Here is the trace

task_id hash    native_id       name    status  exit    submit  duration        realtime        %cpu    rss     vmem    rchar   wchar
1       34/8820a1       102444  Create_Work_Directory   COMPLETED       0       2019-02-07 15:51:06.061 196ms   6ms     0.0%    0       0       0       0
2       f5/6e46bb       102505  Shorter_Data (1)    COMPLETED       0       2019-02-07 15:51:06.345 10.6s   8.6s    99.9%   16.8 MB 328 MB  255.5 MB        236.2 MB
3       2a/087b17       686979  Long_run (1)   COMPLETED       0       2019-02-07 15:51:17.003 5m 4s   4m 59s  1676.9% 4.3 GB  7.3 GB  762.2 MB        47.4 MB
4       27/a49cf2       686980  assign_run (1)  FAILED  2       2019-02-07 15:56:21.039 9.9s    403ms   0.0%    1.6 MB  221.9 MB        284.5 KB        2.4 KB

When I resume the workflow two cached versions of the workflow are used.
N E X T F L O W  ~  version 19.01.0
Launching `dada2.nf` [irreverent_laplace] - revision: 33d784cc0e
[warm up] executor > local
[warm up] executor > uge
[34/8820a1] Cached process > Create_Work_Directory
[f5/6e46bb] Cached process > Shorter_Data (1)

But on my longer running HPC process, the process is rerun from start
[34/11d054] Submitted process > Long_run (1)

The only difference that I can detect is that the two shorter processes are not HPC based jobs where as the longer running process is.  I am running this process from a shared file system.

My initial run command was: 
/home/tallen/nextflow run algorithm.nf  -c example_algorithm_workflow.config   -with-trace

My resume command is the following:
nextflow run algorithm.nf  -c example_algorithm_workflow.config   -with-trace -resume

I am running this on a Redhat based HPC.

I hope that is enough info.

Thanks!

**Comment by pditommaso** [comment]
> (Provide a test case that reproduce the problem either with a self-contained script or GitHub repository)

No test case => no help

**Comment by stoneg64** [comment]
Ok.  The test case would be to run a workflow an UGE process successfully.  The workflow should fail after the UGE process.  Resume the workflow.  The test is successful if the cached version of the UGE process is used when the nextflow is resumed.

Does that work?  Thanks for the help.

**Comment by pditommaso** [comment]
Sorry, I don't wont to be impolite, but for test case I mean a script that can be execute replicating the problem you are reporting.

**Comment by stoneg64** [comment]
In writing the test case I found that I could at least resume some workflows HPC workflows without a problem.  I changed the caching to lenient and my original issue seemed to go away.  User error as usual.  I guess if I had one request it would be more documentation on how to write resumable workflows correctly for different environments.  I see some info here and on the mailing list but a little more in the core documentation would be helpful.  Thanks for the help!
